AnkiDroid/src/main/java/com/ichi2/libanki/Model.java

```diff
@@ -16,6 +34,43 @@ public Model(String json) {
         super(json);
     }
 
+    /** Empty cache. Should be called when the model is saved in the collection. */
+    protected void reset() {
+        mFieldMap = null;
+        mFieldWithCloze = null;
+    }
+
+    /**
+     * @return Mapping of field name -> (ord, field)*/
+    @NonNull
+    public synchronized Map<String, Pair<Integer, JSONObject>> fieldMap() {
+        if (mFieldMap == null) {
+            JSONArray flds = getJSONArray("flds");
+            // TreeMap<Integer, String> map = new TreeMap<Integer, String>();
+            mFieldMap = new HashMap<>();
+            for (JSONObject f: flds.jsonObjectIterable()) {
+                mFieldMap.put(f.getString("name"), new Pair<>(f.getInt("ord"), f));
+            }
+        }
+        return mFieldMap;
+    }
+
+
+    /** Name of the fields which may contain cloze number */
+    public java.util.Collection<String> getFieldNamesWithCLoze() {
```

Comment:
Maybe a bit verbose. Needs a lowercase 'L' in Cloze

```suggestion
    public java.util.Collection<String> getNamesOfFieldsContainingCloze() {
```